src/main/scala/com/ambiata/poacher/hdfs/Hdfs.scala

```diff
@@ -234,7 +234,10 @@ object Hdfs {
     })
 
   def mv(src: Path, dest: Path): Hdfs[Path] = for {
-    fs <- filesystem
+    fs      <- filesystem
+    parent   = dest.getParent
+    e       <- if(fs.getScheme == "file") exists(parent) else ok(true)
+    _       <- unless(e)(fail(s"Parent dir '${dest}' does not exist!"))
```

Comment:
The general decision elsewhere has always been to mkdirs on the parent to ensure it exists.

Prior art: https://github.com/ambiata/mundane/blob/master/mundane-io/src/main/scala/com/ambiata/mundane/io/Files.scala#L42 

Copy should probably do the same IMO.